Add Flutter network headers capture documentation (#37440)
DataDog/documentation: layouts/shortcodes/mdoc/en/sdk/advanced_config/flutter.mdoc.md

- `DatadogRumConfiguration.traceSampleRate` sets a default sampling rate of 20%. If you want all resources requests to generate a full distributed trace, set this value to `100.0`.

### Capture resource headers

> When [tracking resources automatically][10], you can capture HTTP request and response headers on RUM Resources by setting `trackResourceHeaders` on `DatadogRumConfiguration`. This option applies to all Datadog HTTP tracking clients (Tracking HTTP Client, `DatadogClient`, Dio interceptor, and GQL Link). This option is disabled by default.

Reviewer: just wanted to check if `DatadogClient` is the correct name here, or if it should actually be `Datadog Tracking HTTP Client`.

Author: It is, they are 2 distinct clients.

> When [tracking resources automatically][10], you can capture HTTP request and response headers on RUM Resources by setting `trackResourceHeaders` on `DatadogRumConfiguration`. This option applies to all Datadog HTTP tracking clients (Tracking HTTP Client, `DatadogClient`, Dio interceptor, and GQL Link). This option is disabled by default.

Reviewer: ```suggestion
When [tracking resources automatically](#automatically-track-resources), you can capture HTTP request and response headers on RUM Resources by setting `trackResourceHeaders` on `DatadogRumConfiguration`. This option applies to all Datadog HTTP tracking clients (Tracking HTTP Client, `DatadogClient`, Dio Interceptor, and GQL Link). This option is disabled by default.
```

Author: - I think we could keep this `[10]` as it's coherent with the rest of the file, and takes customers to the pub.dev package page.

- Fixed the capitalization on `Interceptor`, thank you.

> When [tracking resources automatically][10], you can capture HTTP request and response headers on RUM Resources by setting `trackResourceHeaders` on `DatadogRumConfiguration`. This option applies to all Datadog HTTP tracking clients (Tracking HTTP Client, `DatadogClient`, Dio interceptor, and GQL Link). This option is disabled by default.

Reviewer: also, what about gRPC? is that supported? it is mentioned on line 471.

Author: For this specific feature it's not supported/applicable. Added a note to clarify it's not included.

Captured headers appear on the RUM Resource event under `resource.request.headers` and `resource.response.headers`. You can query them in the RUM Explorer.

```dart
DatadogRumConfiguration(
  applicationId: '<rum-application-id>',

>   trackResourceHeaders: ResourceHeadersExtractor(),

Author: @codex Where did you see the latest is 3.2.1? 3.3.0 is already available.

)
```

With no arguments, `ResourceHeadersExtractor` captures a predefined set of safe headers:

| Direction | Headers |
|-----------|---------|
| Request | `cache-control`, `content-type` |
| Response | `age`, `cache-control`, `content-encoding`, `content-length`, `content-type`, `etag`, `expires`, `server-timing`, `vary`, `x-cache` |

> To capture additional headers on top of the defaults, pass them through `captureHeaders`. To skip the defaults, set `includeDefaults: false`.

Reviewer: ```suggestion
To capture additional headers in addition to the defaults, pass them through `captureHeaders`. To skip the defaults, set `includeDefaults: false`.
```

Author: Change applied, thank you.
